graph/patterns/change-status.md

```diff
@@ -0,0 +1,86 @@
+# Change Status & Action Request
+
+Microsoft Graph API Design Pattern
+
+_The change status pattern is an effective alternative to model the ability to trigger side effects while keeping an agnostic, data driven approach_
+
+## Problem
+
+In some cases, exposed resources can trigger side effects that modify the system behavior. For instance, a [`riskyUser`](https://docs.microsoft.com/en-us/graph/api/resources/riskyuser?view=graph-rest-1.0) can be [`confirmed`](https://docs.microsoft.com/en-us/graph/api/riskyuser-confirmcompromised?view=graph-rest-1.0&tabs=http) or [`dismissed`](https://docs.microsoft.com/en-us/graph/api/riskyuser-confirmcompromised?view=graph-rest-1.0&tabs=http).
+
+These situation are often modeled as a `POST` request to a specific endpoint in the same URI space of the resource, triggering the side effect directly and not returning any data. While it might be a quick and easy solution, there are some considerations:
+
+- The API should be exposing resources and allow their manipulation, while Actions resemble an object in [OOP][1]
+- It couples the request with the side effect, making more difficult to test, log, replay and reason about what is happening in the system
+- It forces the consumer to follow the [OOP][1] approach at the API level. We do not want to pursue any specific style and the developers writing the client might be working in other ways
```

Comment:
Maybe it forces a non-standard pattern, where generally POST is used to create resources?

Reply:
Not necessarily - the problem is that it uses POST to trigger a side effect, while what we want is to create a resource which will trigger the side effect somehow. I can put emphasis on this if you want!

Reply:
Yes I should have been more clear. It's more about traditional use of POST to a resources collection to create a new resource record in the collection - creation of the resource could trigger some other side effects.  This is rather than a POST to an "action" endpoint that itself triggers side effects.

Reply:
Oh I see what you mean now - point out that usually POST is used to create resources, not to trigger side effects.

I have added such point in 7f3a2c4